pkg/bucketaws/setup.go

```diff
@@ -18,64 +18,131 @@ package s3setup
+
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 )
 
+func ProcessDate(fromDate time.Time) []string {
+
+	// This function formats all dates from provided date
+	// until current date and returns a string formatted
+	// for s3 keys.
+
+	start := fromDate
+	end := time.Now()
+	var dates []string
+	for d := start; !d.After(end); d = d.AddDate(0, 0, 1) {
+		year, month, day := d.Date()
+		dates = append(
+			dates,
+			fmt.Sprintf(
+				"created_year=%d/created_month=%d/created_day=%d",
+				year, month, day))
+	}
+	return dates
+}
+
 func ConfigRead() error {
+
 	// Setting up file formatting
-	// Pulling from Viper
-	viper.SetConfigFile("config/syncron.yaml")
+	// Using Viper
+	// Reading from file syncron.yaml
+
+	viper.AddConfigPath("./config")
+	viper.SetConfigFile("syncron.yaml")
 	viper.SetConfigType("yaml")
-	viper.AddConfigPath(".")
+	viper.SetDefault("download_dir", "/tmp/syncron/")
+
 	// Reading from file
 	err := viper.ReadInConfig()
 	if err != nil {
-		return err
-	} else {
-		logrus.Info("Your configuration file was read succesfully")
-		logrus.Info("Reading from bucket: ", viper.Get("bucket"))
+		logrus.Error(err)
+		logrus.Exit(1)
 	}
+	logrus.Info("Your configuration file was read succesfully")
+	logrus.Info("Reading from bucket: ", viper.Get("bucket"))
 	return nil
 }
 
 func SetupSession() *session.Session {
+
 	// Initialize a session with AWS SDK
-	// It will read from the file located at ~/.aws/credentials and syncron/config/syncron.yaml
+	// It will read from the file located at ~/.aws/credentials and syncron/syncron.yaml
+
 	sess, err := session.NewSession(&aws.Config{
 		Region:   aws.String(viper.GetString("s3.region")),
 		Endpoint: aws.String(viper.GetString("s3.endpoint")),
 	},
 	)
 	if err != nil {
-		fmt.Println("There was an error setting up your aws session")
+		fmt.Println("There was an error setting up your aws session", err)
 		os.Exit(1)
-	} else {
-		logrus.Info("Your AWS session was set up correctly")
 	}
+	logrus.Info("Your AWS session was set up correctly")
 	return sess
 }
 
-func AccessBucket(sess *session.Session) error {
+func AccessBucket(sess *session.Session) (*s3.S3, *s3manager.Downloader) {
+
+	// This function initiates the service for downloading files in s3
+
 	logrus.Info("Accessing bucket...")
 	svc := s3.New(sess)
-	resp, err := svc.ListObjectsV2(&s3.ListObjectsV2Input{Bucket: aws.String(viper.GetString("bucket"))})
-	if err != nil {
-		fmt.Println("Unable to list items in bucket", resp)
-		return err
-	} else {
-		logrus.Info("Success getting into the bucket!")
-	}
-	return nil
+	dwn := s3manager.NewDownloader(sess)
+
+	return svc, dwn
 }
 
-func DownloadFromBucket() error {
+func DownloadFromBucket(svc *s3.S3, dwn *s3manager.Downloader, dates []string, bprefix string) error {
```

Comment:
Instead of passing a complicated to generate argument like 'dates', why don't you just pass the time here and process it inside instead?